### PR TITLE
SPOI-4136 Adding loadingPromise to table options

### PR DESCRIPTION
**Pull request by sashadt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,16 +70,18 @@ The options object should be available on the parent scope of the `<mlhr-table>`
...
 | fixedHeight        | `boolean` | false       | If true, the table body will always have a height of `bodyHeight`, regardless of whether the rows fill up the vertical space.        |
+| onRegisterApi      | `function` | {}         | Provides a access to select table controller methods, including selectAll, deselectAll, isSelectedAll, setLoading, etc. | 
 
 
 ### Loading
-A common requirement for tables showing dynamically loaded data is to show loading feedback. There are several options pertaining to this: `loading`, `loadingText`, and `loadingTemplateUrl`.
+A common requirement for tables showing dynamically loaded data is to show loading feedback. There are several options pertaining to this: `loading`, `loadingText`, and `loadingTemplateUrl`.  To disable loading text, a promise object from data loading can be provided, so that `setLoading(false)` can be attached to `promise.then()`.  Optionally, `onRegisterApi` function can be specified, which provides direct access to `setLoading` and other table controller methods.  This function specifies a single argument, which is the api object provided by the table.  Example: `onRegisterApi: function(api) { $scope.tableAPI = api; }`.
```
Comment (andyperlitch) [inline README.md]: You should probably add loadingPromise to that list of options pertaining to loading. I see that you are describing the `loadingPromise` option without actually mentioning it by name.

File: src/directives/mlhrTable.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ angular.module('datatorrent.mlhrTable.directives.mlhrTable', [
         'glyphicon glyphicon-chevron-down'
       ],
       onRegisterApi: function(api) {
-        // noop
+        // noop - user overrides and gets a hold of api object
```
Comment (andyperlitch) [inline src/directives/mlhrTable.js]: nice touch

**Comment by andyperlitch** [comment]
merged, if you want to update the readme, just push directly to master

File: README.md (hunk 1 above)
Comment (sashadt) [inline README.md]: It's in there already from earlier fixes.  See https://github.com/DataTorrent/malhar-angular-table

**Pull request merged**